Python/ast_unparse.c

```diff
@@ -67,6 +86,9 @@ append_repr(_PyUnicodeWriter *writer, PyObject *obj)
     if (!repr) {
         return -1;
     }
+    if ((PyFloat_CheckExact(obj) || PyComplex_CheckExact(obj)) && !(repr = replace_inf_value(repr))) {
```

Comment:
There is a reference leak: you must add Py_XDECREF(repr).

Reply:
Can you clarify the case? If it drops to `return -1` block isn't `repr` itself already resulted with `NULL`?